doc/buffer.md

```diff
@@ -1,5 +1,144 @@
+- `[in] length`: The number of `T` elements to allocate.
+
+Returns a new `Buffer` object.
+
+### New
+
+Wraps the provided external data into a new `Buffer` object.
+
+The `Buffer` object does not assume ownership for the data and expects it to be
+valid for the lifetime of the object. Since the `Buffer` is subject to garbage
+collection this overload is only suitable for data which is static and never
+needs to be freed.
+
+```cpp
+static Buffer<T> New(napi_env env, T* data, size_t length);
+```
+
+- `[in] env`: The environment in which to create the `Buffer` object.
+- `[in] data`: The pointer to the external data to expose.
+- `[in] length`: The number of `T` elements in the external data.
+
+Returns a new `Buffer` object.
+
+### New
+
+Wraps the provided external data into a new `Buffer` object.
+
+The Bu`Buffer`ffer object does not assume ownership for the data and expects it
+to be valid for the lifetime of the object. The data can only be freed once the
+`finalizeCallback` is invoked to indicate that the `Buffer` has been released.
+
+```cpp
+template <typename Finalizer>
+static Buffer<T> New(napi_env env,
+                     T* data,
+                     size_t length,
+                     Finalizer finalizeCallback);
+```
+
+- `[in] env`: The environment in which to create the `Buffer` object.
+- `[in] data`: The pointer to the external data to expose.
+- `[in] length`: The number of `T` elements in the external data.
+- `[in] finalizeCallback`: The function to be called when the `Buffer` is
+  destroyed. It must implement `operator()`, accept a `T*` (which is the
+  external data pointer), and return `void`.
+
+Returns a new `Buffer` object.
+
+### New
+
+Wraps the provided external data into a new `Buffer` object.
+
+The `Buffer` object does not assume ownership for the data and expects it to be
+valid for the lifetime of the object. The data can only be freed once the
+`finalizeCallback` is invoked to indicate that the `Buffer` has been released.
+
+```cpp
+template <typename Finalizer, typename Hint>
+static Buffer<T> New(napi_env env,
+                     T* data,
+                     size_t length,
+                     Finalizer finalizeCallback,
+                     Hint* finalizeHint);
+```
+
+- `[in] env`: The environment in which to create the `Buffer` object.
+- `[in] data`: The pointer to the external data to expose.
+- `[in] length`: The number of `T` elements in the external data.
+- `[in] finalizeCallback`: The function to be called when the `Buffer` is
+  destroyed. It must implement `operator()`, accept a `T*` (which is the
+  external data pointer) and `Hint*`, and return `void`.
+- `[in] finalizeHint`: The hint to be passed as the second parameter of the
+  finalize callback.
+
+Returns a new `Buffer` object.
+
+### Copy
+
+Allocates a new `Buffer` object and copies the provided external data into it.
+
+The `Buffer` object does not assume ownership for the data and expects it to be
```

Comment:
I don't get it. If it copies the data, how does it not assume ownership of it.

Reply:
The Buffer doesn't assume ownership for the data you're providing to the `Copy` call, it copies it to memory that it does own.  This was supposed to contrast with the other Buffer factory functions, but maybe it's not necessary?